draft-ietf-quic-http.md

```diff
@@ -496,35 +496,26 @@ HEADERS frames can only be sent on request / push streams.
 ### PRIORITY {#frame-priority}
 
 The PRIORITY (type=0x2) frame specifies the client-advised priority of a
-request, server push or placeholder.
-
-A PRIORITY frame identifies an element to prioritize, and an element upon which
-it depends.  A Prioritized ID or Dependency ID identifies a client-initiated
-request using the corresponding stream ID, a server push using a Push ID (see
-{{frame-push-promise}}), or a placeholder using a Placeholder ID (see
-{{placeholders}}).
-
-When a client initiates a request, a PRIORITY frame MAY be sent as the first
-frame of the stream, creating a dependency on an existing element.  In order to
-ensure that prioritization is processed in a consistent order, any subsequent
-PRIORITY frames for that request MUST be sent on the control stream.  A
-PRIORITY frame received after other frames on a request stream MUST be treated
+request or server push.
+
+A PRIORITY frame identifies an element to prioritize.  A Prioritized ID identifies
+a client-initiated request using the corresponding stream ID or a server push using
+a Push ID (see {{frame-push-promise}}).
+
+In order to ensure that prioritization is processed in a consistent order
+PRIORITY frames MUST be sent on the control stream.  A
+PRIORITY frame received on any other stream MUST be treated
 as a stream error of type HTTP_UNEXPECTED_FRAME.
 
-If, by the time a new request stream is opened, its priority information
-has already been received via the control stream, the PRIORITY frame
-sent on the request stream MUST be ignored.
 
 ~~~~~~~~~~  drawing
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|PT |DT | Empty |         [Prioritized Element ID (i)]        ...
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                [Element Dependency ID (i)]                  ...
+|PT |   Empty   |         [Prioritized Element ID (i)]        ...
```

Comment:
Prioritized Element ID is optional here.  What happens if it's absent?  If you allowed it on the request stream, then absent would presumably mean "this stream"?

Reply:
The text currently reads that it is only allowed on the control stream, where it would be required. Do you think that should be changed?

Reply:
Nope, I think that's ok, but the [] around Prioritized Element ID (i) implies it's optional.  But if you're sending it on the control stream, I think it's always required?